### ISSUE TITLE
Devtools tab not showing

### ISSUE THREAD

**Issue by mackelito** [angular/angular#54190]
### Is this a bug report or a feature request?

- [X] Bug Report
- [ ] Feature Request

### Please provide the steps to reproduce the issue [Bug Report only]

I can't see the tab in chrome devtools..
I even assigned a shortcut for it and when I use it I get a dialog that says
```
Angular application running development mode.
Open developer tools, and select the Angular tab.
```

But it's still not there

### Please provide the expected behavior vs the actual behavior you encountered [Bug Report only]

the tab should show :P

**Comment by JeanMeche** [comment]
![Screenshot 2024-02-01 at 01 35 02](https://github.com/angular/angular/assets/1300985/14924ca4-0142-4391-bf52-4bff370b6bef)

Do you have it maybe in the expanded menu ?

**Comment by mackelito** [comment]
Nope...
![Screenshot 2024-02-01 at 10 04 39](https://github.com/angular/angular/assets/598463/e1d582e3-7638-439a-af72-b8a568b792f5)

**Comment by JoostK** [comment]
Is it an Incognito tab?

**Comment by zeljkoantich** [comment]
Yeah, I had the same issue on Linux _Google Chrome Version 121.0.6167.139 (Official Build) (64-bit)_.

I got the tab back, after removing/reinstalling Angular DevTools + Browser restart.

**Comment by mackelito** [comment]
> Yeah, I had the same issue on Linux _Google Chrome Version 121.0.6167.139 (Official Build) (64-bit)_.
> 
> I got the tab back, after removing/reinstalling Angular DevTools + Browser restart.

I have tried to reinstall and restart so many times now :D

**Comment by mackelito** [comment]
> Is it an Incognito tab?

No.. not incognito.. but I did try there as well (after enabled in incognito)

**Comment by mrByco** [comment]
Same issue

**Comment by JeanMeche** [comment]
Could you check if of the extension page you have an error button for the Angular devtools ? 

![Screenshot 2024-02-02 at 19 02 12](https://github.com/angular/angular/assets/1300985/12fecc19-f08f-47ee-b01f-4f8406c4d996)

**Comment by mackelito** [comment]
Sorry for the late reply.. no errors at all
![Screenshot 2024-02-05 at 13 36 07](https://github.com/angular/angular/assets/598463/e7d34a45-0121-4f6b-a157-3e6ecece2539)

**Comment by AhsanAyaz** [comment]
I also don't see the Angular tab in the chrome devtools. 
![image](https://github.com/angular/angular/assets/9844254/0ee3adac-02da-4627-b18e-64f8cd7b51f5)

**Comment by markusahrweileramcon** [comment]
same here

**Comment by DmitryAvanesov** [comment]
same issue

**Comment by eneajaho** [comment]
same issue, no error, just doesn't show up

**Comment by jamesdanielspassle** [comment]
Same issue. I uninstalled and reinstalled dev tools. All I had to do was close and re-open the tab. I didn't need to close Chrome or restart my machine. I'm not even sure I needed to uninstall DevTools

**Comment by cmrabet** [comment]
Same issue here; Chrome Version 123.0.6312.58.

Re-installed, restarted, etc.. many times. Tab doesn't show up, entry menu doesn't show up under more tools either.

**Comment by fonnerk** [comment]
Same issue here: Chrome Version 123.0.6312.86

Reinstalled, restarted.  Tab does not show up in developer tools... Interestingly on another system of mine running chrome it worked fine...

Also of note... the pinned extension DOES show up when hitting my angular site... so the extension is running to a degree... it detects their is an angular site.  The Angular tab in the dev tools window is just not showing up. 

while troubleshooting... I turned on developer mode for extensions... which gave me a setting for the extension to collect errors...  I turned that on in angular dev tools and after loading my test app, I saw an errors button on the extension.

Error displayed in the errors screen for the angular dev tools extension...
=====================================
Uncaught Error: Extension context invalidated.

Context
http://localhost:4301/

Stack Trace
app/ng_validate_bundle.js:1 (anonymous function)

"use strict";(()=>{window.addEventListener("message",e=>{e.source===window&&e.data&&chrome.runtime.sendMessage(e.data)});if(document.contentType==="text/html"){let e=document.createElement("script");e.src=chrome.runtime.getURL("app/detect_angular_for_extension_icon_bundle.js"),document.documentElement.appendChild(e),document.documentElement.removeChild(e)}})();
=====================================

Let me know if I can provide anything else that would help.

**Comment by rockit666** [comment]
I created a new Angular 17 app. Try Dev Tools, looks weird, then Dev Tools crash and must repaired.  On Console I get this message. This message is in a loop.

![image](https://github.com/angular/angular/assets/38085581/516f0297-be79-4fc2-b7e9-9b6d8a4e2f32)
![image](https://github.com/angular/angular/assets/38085581/c56e656e-c541-4382-a2c5-1bbe64e92b5f)
![image](https://github.com/angular/angular/assets/38085581/3e4e1376-6555-4ff9-9356-3bc86ac1f852)

**Comment by jbjhjm** [comment]
Note that besides the tab not showing up, the extension may slow down your devtools extremely. My devtools were lagging, any console logging was delayed by 2-5 seconds til showing up... well, until I uninstalled angular devtools - and all of a sudden, performance returned. Not a good day for ng devtools.

**Comment by localpcguy** [comment]
I've seen the same error shown above by @fonnerk and @rockit666 - I can specifically get it to trigger by going into the extension and clicking to enable file urls. The error starts looping immediately, but after a page refresh it does not seem to reappear. Regardless of that setting the extension never shows a tab in the inspector / dev tools. 
<img width="779" alt="image" src="https://github.com/angular/angular/assets/593240/169aa84a-0f24-4e9a-8c2b-3d2e6cc8fa84">
 
I originally thought it was due to a fairly restrictive Content Security Policy, but I removed that temporarily and it did not make a difference in the extension's ability to initialize (plus, I would expect to see errors in the console stating the extension scripts were blocked by the CSP and that doesn't happen).
